package/src/utils/add-virtual-import.ts

```diff
@@ -0,0 +1,71 @@
+import type { HookParameters } from "astro";
+import { type Plugin } from "vite";
+
+function resolveVirtualModuleId<T extends string>(id: T): `\0${T}` {
+	return `\0${id}`;
+}
+
+function createVirtualModule(name: string, content: string): Plugin {
+	const pluginName = `vite-plugin-${name}`;
```

Comment:
Is that safe enough? Maybe we also need the integration name for safety?

Reply:
Well `name` here _should_ be something like `virtual:my-integration/config` so it should be safe enough?

Reply:
yeah you're right actually